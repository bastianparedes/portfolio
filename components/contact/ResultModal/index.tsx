import React from 'react';

import { Modal } from 'bastianparedes/components';
import type { IconType } from 'react-icons';

import styles from './styles.module.scss';

interface props {
  setModalVisible: (boolean: boolean) => void;
  Icon?: IconType | null;
  text?: string;
}

const ResultModal = ({
  setModalVisible,
  Icon = null,
  text = ''
}: props): JSX.Element => {
  return (
    <Modal setModalVisible={setModalVisible}>
      <div className={styles.container}>
        <div className={styles.iconContainer}>{Icon !== null && <Icon />}</div>
        <div className={styles.textContainer}>
          <p className={styles.text}>{text}</p>
        </div>
      </div>
    </Modal>
  );
};

export default ResultModal;
