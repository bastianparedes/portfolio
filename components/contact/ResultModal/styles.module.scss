.container {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 2rem;
}

.iconContainer {
  width: fit-content;
  height: fit-content;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 4rem;
}

.textContainer {
  position: relative;
}

.text {
  font-size: 1.5rem;
  text-align: justify;
}
