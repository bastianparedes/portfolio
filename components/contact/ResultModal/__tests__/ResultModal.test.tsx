import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { GrStatusGood } from 'react-icons/gr';

import ResultModal from '..';
jest.mock('bastianparedes/components', () => ({
  Modal: ({
    setModalVisible,
    children
  }: {
    setModalVisible: (boolean: boolean) => void;
    children: React.ReactNode;
  }) => {
    return (
      <>
        <button
          onClick={() => {
            setModalVisible(false);
          }}
        >
          X
        </button>
        {children}
      </>
    );
  }
}));

describe('<ResultModal />', () => {
  it('should render', () => {
    const props = {
      Icon: GrStatusGood,
      setModalVisible: jest.fn(),
      text: 'Success'
    };

    const { container } = render(<ResultModal {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without props', () => {
    const props = {
      setModalVisible: jest.fn()
    };

    const { container } = render(<ResultModal {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should close modal when button is clicked', async () => {
    const props = {
      Icon: GrStatusGood,
      setModalVisible: jest.fn(),
      text: 'Success'
    };

    const { getByText } = render(<ResultModal {...props} />);
    const button = getByText('X');
    act(() => {
      fireEvent.click(button);
    });

    expect(props.setModalVisible).toBeCalledTimes(1);
    expect(props.setModalVisible).toBeCalledWith(false);
  });
});
