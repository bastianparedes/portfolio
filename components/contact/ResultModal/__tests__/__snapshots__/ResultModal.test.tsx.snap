// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultModal /> should render 1`] = `
<div>
  <button>
    X
  </button>
  <div
    class="container"
  >
    <div
      class="iconContainer"
    />
    <div
      class="textContainer"
    >
      <p
        class="text"
      >
        Success
      </p>
    </div>
  </div>
</div>
`;

exports[`<ResultModal /> should render without props 1`] = `
<div>
  <button>
    X
  </button>
  <div
    class="container"
  >
    <div
      class="iconContainer"
    />
    <div
      class="textContainer"
    >
      <p
        class="text"
      />
    </div>
  </div>
</div>
`;
